x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizerTests.java

```diff
@@ -3237,6 +3239,162 @@ public void testStatsWithCanonicalAggregate() throws Exception {
+        var exprs = eval.fields();
+
+        // s = count(*) * 3
+        var mul = as(Alias.unwrap(exprs.get(0)), Mul.class);
+        assertThat(mul.source().text(), equalTo("sum([1,2])"));
+        var left = as(mul.left(), ReferenceAttribute.class);
+        assertThat(left.name(), equalTo("$$COUNT$s$0"));
+        var right = as(mul.right(), Literal.class);
+        assertThat(right.value(), equalTo(3));
+
+        // s_expr = count(*) * 3.14
+        var mul_expr = as(Alias.unwrap(exprs.get(1)), Mul.class);
+        assertThat(mul_expr.source().text(), equalTo("sum(314.0/100)"));
+        var left_expr = as(mul_expr.left(), ReferenceAttribute.class);
+        assertThat(left_expr.name(), equalTo("$$COUNT$s$0"));
+        var right_expr = as(mul_expr.right(), Literal.class);
+        assertThat(right_expr.value(), equalTo(3.14));
+
+        // s_null = null
+        var s_null = as(Alias.unwrap(exprs.get(2)), Literal.class);
+        assertThat(s_null.source().text(), equalTo("sum(null)"));
+        assertThat(s_null.value(), equalTo(null));
+
+        var limit = as(eval.child(), Limit.class);
+        var agg = as(limit.child(), Aggregate.class);
+        var count = as(Alias.unwrap(agg.aggregates().get(0)), Count.class);
+        assertThat(count.children().get(0), instanceOf(Literal.class));
+        var w = as(Alias.unwrap(agg.groupings().get(0)), ReferenceAttribute.class);
+        assertThat(w.name(), equalTo("w"));
+    }
+
+    private record AggOfLiteralTestCase(String aggFunctionName, Function<int[], Object> aggMultiValue) {};
+
+    /**
+     * Aggs of literals in case that the agg can be simply replaced by a corresponding mv-function;
+     * e.g. avg([1,2,3]) which is equivalent to mv_avg([1,2,3]).
+     *
+     * Expects e.g.
+     *
+     * EsqlProject[[s{r}#3, s_expr{r}#5, s_null{r}#7]]
+     * \_Limit[1000[INTEGER]]
+     *   \_Row[[1.5[DOUBLE] AS s, 3.14[DOUBLE] AS s_expr, null[DOUBLE] AS s_null]]
+     */
+    public void testAggOfLiteral() {
+        List<AggOfLiteralTestCase> cases = List.of(
+            new AggOfLiteralTestCase("avg", ints -> ((double) Arrays.stream(ints).sum()) / ints.length),
+            new AggOfLiteralTestCase("min", ints -> Arrays.stream(ints).min().getAsInt()),
+            new AggOfLiteralTestCase("max", ints -> Arrays.stream(ints).max().getAsInt())
+        );
+
+        for (AggOfLiteralTestCase testCase : cases) {
+            String query = LoggerMessageFormat.format(null, """
+                from test
+                | stats s = {}([1,2]),
+                        s_expr = {}(314.0/100),
+                        s_null = {}(null)
+                | keep s, s_expr, s_null
+                """, testCase.aggFunctionName, testCase.aggFunctionName, testCase.aggFunctionName);
+
+            var plan = optimizedPlan(query);
+
+            var project = as(plan, Project.class);
+            var limit = as(project.child(), Limit.class);
+            var row = as(limit.child(), Row.class);
```

Comment:
This is probably an error here with `row`.

Reply:
Discussed with Andrei:
- We both thought that it would be nicer to use a `LocalRelation` with 0 attributes and a subsequent `Eval` on top 
 that contains `s = mv_avg(...), ...` - esp. given that `Row` is mostly used in tests and not production.
- It turns out that `LocalRelation` with 0 attributes is not correct, though, as this would automatically have 0 rows as well; it seems that `Row` is the most fitting concept, here, after all.

Reply:
To stick closer to the original code in `SubstituteSurrogates`, I handle this case now by
- creating a placeholder `Row` when we end up not having any actual aggs, and
- handling the evaluation of `s = mv_avg(...)` in a subsequent `eval` - this required less change, actually, as that's how we handle the `avg` surrogate substitution today.